internal/clients/embed.go

```diff
@@ -0,0 +1,54 @@
+package clients
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+var AwsSupported = make(map[string]interface{})
+var ErrArchitectureNotSupported = errors.New("architecture is not supported")
+
+const x86_64 = "x86_64"
+const arm64 = "arm"
+const i386 = "i386"
+
+type supportedInstanceTypes struct {
+	Aws struct {
+		Ec2 []string `yaml:"ec2"`
+	} `yaml:"aws"`
+}
+
+//go:embed supported_instance_types.yml
+var f embed.FS
+
+func init() {
+	var supported supportedInstanceTypes
+	it, err := f.ReadFile("supported_instance_types.yml")
+	if err != nil {
+		panic(err)
+	}
+
+	if err := yaml.Unmarshal(it, &supported); err != nil {
+		panic(err)
+	}
+
+	for _, i := range supported.Aws.Ec2 {
+		AwsSupported[i] = nil
+	}
+}
+
+func MapArchitectures(ctx context.Context, arch string) (string, error) {
+	switch {
+	case strings.Contains(arch, "i386"):
+		return i386, nil
+	case strings.Contains(arch, "86"):
+		return x86_64, nil
+	case strings.Contains(arch, "arm"):
+		return arm64, nil
+	}
+	return "", ErrArchitectureNotSupported
+}
```

Comment:
I believe this mapping is ec2 only, so move this into ec2/ package.

Reply:
Why? Cant it be suitable for Azure as well?

Reply:
True, we can convert different labels, yeah. Let's keep it, sure.